packages/execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -41,6 +43,16 @@ const fallbackError = {
   message: 'Execution Environment Error',
 };
 
+export type InvokeSnapArgs = Parameters<
+  OnRpcRequestHandler | GetTransactionInsightHandler
```

Comment:
Is it possible to use `SnapExports` here instead of explicitly stating the handlers? I can see this getting growing as we add more handlers in the future.

Reply:
Maybe, we might need some TypeScript magic to extract the parameters from each value of `SnapExports` though.

Reply:
https://www.typescriptlang.org/play?#code/PTAEAUBsEMGMFMAWB7SATeAnUAXAngA7wBQ+RoAUgM7IB2ASgbPfAI4Cu8VOAPACqF4APlABeUAKIBuYsRATBVUNCUYAZgEta8NKC2gAyrWgEqpQYeMFGsABLRaaSFjGgAFNEwBzKgC5QAN7EoKDImBpeWv7c4bReMiGYbJzc-tR0NiwcXLzstADWtMgA7rQA2gC6oAA+gaBl+fB40TixXhX+eYUltKAAvkIyfQCUYiLgmMgAthpU8DxdRaWDsubkAPIMTFkpOPaOztjiRiY2+05YMmvwoADi8Dh8mA5UcDgadACStFQRiHsOC5HdyeHz+IIhMIRKKgGJaeLBXDPH5vD60cH1RrNWGteEdUCLHr9IajUTjSYzOY8OFxFayMg3E4EACiAA8CGEcEpxBDQltmMkcgB+fybTKC7jnQ4JUBeB5PF6or4-P44EV3eXI16wd7K35ef5Sy7EPpXeS2eCQIjYBm4ZCgeCs1pvUAETzQKYPLBKNQU5S9ZAAIwAVvAdddQOsQ2GcOB3Z6cN7+BZHYnHEoWLAwmhqbi4gAadwAOhLoL8-rwlVJIkJyxE4jjzwTSck8AaTWQagURAqdPkfEQNzdTa9mCUndwg9C0Z1oDUeR1aKoRYjTLZHMwXMbHtH3MjM9j8d3PDX7M5VBWQA

Reply:
yes! 😄